CTLD.lua

```diff
@@ -7187,12 +7165,88 @@ function ctld.initialize(force)
     end
     env.info("END search for crates")
 
+    -- register event handler
+    ctld.logInfo("registering event handler")
+    world.addEventHandler(ctld.eventHandler)
+
+
     -- don't initialize more than once
     ctld.alreadyInitialized = true
 
     env.info("CTLD READY")
 end
 
+--- Handle world events.
+ctld.eventHandler = {}
+function ctld.eventHandler:onEvent(event)
+    if event == nil then
+        ctld.logError("Event handler was called with a nil event!")
+        return
+    end
+
+    -- check that we know the event
+    if not (
+        event.id == 15 -- S_EVENT_BIRTH"
+        or event.id == 20 -- S_EVENT_PLAYER_ENTER_UNIT
+        ) then
+        return
+    end
+
+    -- find the originator unit
+    local unitName = nil
+    if event.initiator ~= nil and event.initiator.getName then
+        unitName = event.initiator:getName()
+    end
+    if not unitName then 
+        ctld.logWarning("no unitname found in event %s", ctld.p(event))
+        return
+    end
+
+    if mist.DBs.humansByName[unitName] then -- it's a human unit
+        ctld.logDebug("caught event BIRTH for human unit [%s]", ctld.p(unitName))
+        local _unit = Unit.getByName(unitName)
+        if _unit ~= nil then
+            -- assign transport pilot
+            ctld.logTrace("_unit = %s", ctld.p(_unit))
+
+            local playerTypeName = _unit:getTypeName()
+            ctld.logTrace("playerTypeName = %s", ctld.p(playerTypeName))
+
+            local notFound = true
+
+            for _,transportPilotName in pairs(ctld.transportPilotNames) do
+                if transportPilotName == unitName then
+                    ctld.logDebug("Unit is already a transport pilot, skipping...")
+                    notFound = false
+                    break
+                end
+            end
```

Comment:
This block is preventing the F10 menu creation for any of the predefined transport names since they aren't initialized anywhere else.  It doesn't look like the notFound stuff is needed at all.

Reply:
I love the concept though, much better to be setting those menus when  needed, not just on a timer that runs all the time.  Once I took that block out it worked fine.